### ISSUE TITLE
uBO responseheader

### ISSUE THREAD

**Issue by scripthunter7** [AdguardTeam/AGLint#2]
The parser handles all ADG / ABP / uBO rules except the following rule:

```adblock
example.com##^responseheader(something)
```

Syntactically, this follows the pattern of a cosmetic rule, but technically it is a network rule... and AdGuard - using the right approach - considers it as a network rule:

```adblock
||example.com^$removeheader=something
```

Currently, the parser detects it at the cosmetic level, but simply returns null to it:
https://github.com/AdguardTeam/AGLint/blob/e8f2b8d81562a6c13b2198e62e488872ae01c202/src/parser/cosmetic/cosmetic.ts#L357-L361

What would be the correct behavior here? Due to the parsing algorithm, the cosmetic rule parser will definitely detect it (since it detects all separators, including `##^`), but it wouldn't be nice if the cosmetic rule parser returned with a network rule ast. If the network parser should detect it again, it would be an unnecessary extra step after that.

One more note about uBO: the legacy `script:inject(...)` is not handled by this parser, but it has been deprecated for a long time as far as I know.

Reference: https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#response-header-filtering
